x-pack/platform/plugins/shared/streams/server/lib/telemetry/stats/streams_usage_collector.ts

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { STREAMS_RULE_TYPE_IDS } from '@kbn/rule-data-utils';
+import { Streams } from '@kbn/streams-schema';
+import type { UsageCollectionSetup } from '@kbn/usage-collection-plugin/server';
+import type { Logger } from '@kbn/core/server';
+import type { StreamsClient } from '../../streams/client';
+import { hasChangedRetention, percentiles } from './utils';
+import type { StreamsStatsTelemetry } from './types';
+import { registerStreamsUsageCollector as registerCollector } from './register_collector';
+
+function createFetchFunction(
+  logger: Logger,
+  getStreamsClient: () => Promise<StreamsClient>,
+  isDev: boolean
+) {
+  return async function fetchStreamsUsageStats({
+    esClient,
+  }: {
+    esClient: any;
+  }): Promise<StreamsStatsTelemetry> {
+    try {
+      const streamsClient = await getStreamsClient();
+
+      const [allStreamsData, significantEventsMetrics, ruleExecutionMetrics] = await Promise.all([
+        streamsClient.listStreams(),
```

Comment:
`listStreams` is too much, it also returns regular data streams that were never touched via the streams UI/API. What we need are the "managed" streams.

Also, it's not necessary to check access to them, since this is not the point (see https://github.com/elastic/kibana/blob/578b04eb2fe1558fc3e41f9a9811cf03f65e30d0/x-pack/platform/plugins/shared/streams/server/lib/streams/client.ts#L680 which is doing this automatically)